### PR TITLE
Set a cgroup if containerized

### PR DESCRIPTION
**Pull request by Oats87**
When there is a defined cgroup for PID 1, assume we are containerized and set a root

#### Proposed Changes ####
Set `--kubelet-cgroups` and `--runtime-cgroups` on the `kubelet` only if we think we are containerized and running with `systemd`.

#### Types of Changes ####
Bugfix

#### Verification ####
Run `K3s` on a CentOS 7.8+ system and observe that it starts.

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/2548

#### Further Comments ####
This specifically examines the cgroup for PID 1, and makes an educated guess as to whether we should set the `--*-cgroups` flags to `/systemd`

### CODE REVIEW

File: pkg/daemons/agent/agent.go [hunk 1]
```diff
@@ -194,12 +194,27 @@ func checkCgroups() (root string, hasCFS bool, hasPIDs bool) {
...
+	g, err := os.Open("/proc/1/cgroup")
+	if err != nil {
+		return "", false, false
+	}
+	defer g.Close()
+	root = ""
+	scan = bufio.NewScanner(g)
+	for scan.Scan() {
+		parts := strings.Split(scan.Text(), ":")
+		if len(parts) < 3 {
```
Comment (dweomer) [inline pkg/daemons/agent/agent.go:210]: What does this mean? Unexpected formatting? If so, shouldn't log at warning about it? Some other condition we can safely ignore? If so, please explain or link to a control groups doc that explains.

File: pkg/daemons/agent/agent.go [hunk 2]
```diff
@@ -194,12 +194,27 @@ func checkCgroups() (root string, hasCFS bool, hasPIDs bool) {
 				if _, err := os.Stat(p); err == nil {
 					hasCFS = true
 				}
-			} else if system == "name=systemd" {
+			}
+		}
+	}
+
+	g, err := os.Open("/proc/1/cgroup")
```
Comment (brandond) [inline pkg/daemons/agent/agent.go:201]: Can you add a comment explaining why we're checking the init process's cgroups as opposed to our own?

File: pkg/daemons/agent/agent.go (hunk 1 above)
Comment (brandond) [reply pkg/daemons/agent/agent.go:210]: +1 - some inline comments explaining what is being scanned for and why would be helpful.

Comment (Oats87) [reply pkg/daemons/agent/agent.go:210]: This logic was taken from above; I am assuming that in the event of unexpected formatting our check falls through as the system is not formatting the contents of these `proc` files properly

**Review by dweomer — APPROVED** [review]

**Review by briandowns — APPROVED** [review]
LGTM (nits from DM not blocking)

**Pull request merged**